File: src/providers/chatgpt/schemas.ts
/**
 * Zod schemas for raw ChatGPT archive data.
 */
import { z } from "zod";

export const ChatGPTAuthorSchema = z.object({
  role: z.string(),
});

export const ChatGPTContentSchema = z.object({
  content_type: z.string().optional().nullable(),
  parts: z.array(z.string()).optional().nullable(),
});

export const ChatGPTMessageSchema = z.object({
  author: ChatGPTAuthorSchema,
  content: ChatGPTContentSchema,
  create_time: z.number().optional().nullable(),
});

export type ChatGPTAuthor = z.infer<typeof ChatGPTAuthorSchema>;
export type ChatGPTContent = z.infer<typeof ChatGPTContentSchema>;
export type ChatGPTMessage = z.infer<typeof ChatGPTMessageSchema>;

File: context_use/etl/providers/chatgpt/schemas.py
"""Pydantic schemas for raw ChatGPT archive data."""

from __future__ import annotations

from pydantic import BaseModel


class ChatGPTAuthor(BaseModel):
    role: str


class ChatGPTContent(BaseModel):
    content_type: str | None = None
    parts: list[str] | None = None


class ChatGPTMessage(BaseModel):
    author: ChatGPTAuthor
    content: ChatGPTContent
    create_time: float | None = None
